WMIAdventure/backend/WMIAdventure_backend/battle/businesslogic/effects/RandomizeDeckEffect.py

```diff
@@ -5,14 +5,20 @@
 class RandomizeDeckEffect(Effect):
     """
     Randomizes order of cards in deck.
+    This effect randomizes cards order til it's changed, there is no randomization which does nothing.
     """
 
     def on_activation(self, target, turns_queue):
         deck = target.deck
 
-        cards_order = [None for _ in range(deck.size())]
-        for i in range(deck.size()):
-            cards_order[i] = deck.lookup(i)
+        original_cards_order = [i for i in range(deck.size())]
+        new_cards_order = [i for i in range(deck.size())]
 
-        random.shuffle(cards_order)
-        deck.create_cards_queue(tuple(cards_order))
+        # Randomize cards order til it's changed.
+        while new_cards_order == original_cards_order:
+            random.shuffle(new_cards_order)
```

Comment:
Rozumiem założenie - kolejność decku MUSI się zmienić. Ale czy to wtedy jest w pełni randomizacja? Dlaczego nie możemy wylosować takiej samej kolejności kart? To jest ze względu na testowanie, czy po prostu takie założenie podjąłeś?

Reply:
Nie jest to pełna randomizacja, na początku napisałem test, który 100 razy używał efektu i jeśli choć raz zmieniła się kolejność to test przechodził. Potem jakoś to zmieniłem żeby musiała się zmienić kolejność, w sumie bez powodu.